internal/runner/runner.go

```diff
@@ -125,14 +125,6 @@ func New(options *types.Options) (*Runner, error) {
 			}
 		}
 
-		// if template list or template display is enabled, enable all templates
-		if options.TemplateList || options.TemplateDisplay {
-			options.EnableCodeTemplates = true
-			options.EnableFileTemplates = true
-			options.EnableSelfContainedTemplates = true
-			options.EnableGlobalMatchersTemplates = true
-		}
-
```

Comment:
Introduced in #6343. Removed, since mutating global opts in the runner could affects the entire engine state.